fearless/Modules/Staking/Analytics/AnalyticsValidators/View/AnalyticsValidatorsCell.swift

```diff
@@ -0,0 +1,150 @@
+    func didTapInfoButton(in cell: AnalyticsValidatorsCell)
+}
+
+final class AnalyticsValidatorsCell: UITableViewCell {
+    weak var delegate: AnalyticsValidatorsCellDelegate?
+
+    let iconView: PolkadotIconView = {
+        let view = PolkadotIconView()
+        view.backgroundColor = .clear
+        view.fillColor = R.color.colorWhite()!
+        return view
+    }()
+
+    let nameLabel: UILabel = {
+        let label = UILabel()
+        label.font = .p1Paragraph
+        label.textColor = R.color.colorWhite()
+        return label
+    }()
+
+    let progressView: RoundedView = {
+        let view = RoundedView()
+        view.fillColor = R.color.colorAccent()!
+        return view
+    }()
+
+    private var progressPercents: Double = 0.0
+    private var widthConstraint: Constraint?
+
+    let progressValueLabel: UILabel = {
+        let label = UILabel()
+        label.font = .p3Paragraph
+        label.textColor = R.color.colorLightGray()
+        return label
+    }()
+
+    let infoButton: UIButton = {
+        let button = UIButton()
+        button.setImage(R.image.iconInfo(), for: .normal)
+        return button
+    }()
+
+    private lazy var progressStackView: UIStackView = {
+        UIView.hStack(
+            alignment: .center,
+            spacing: 8,
+            [progressView, progressValueLabel]
+        )
+    }()
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+        setupLayout()
+        setupBackground()
+        infoButton.addTarget(self, action: #selector(tapInfoButton), for: .touchUpInside)
+    }
+
+    @available(*, unavailable)
+    required init?(coder _: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func prepareForReuse() {
+        super.prepareForReuse()
+
+        progressPercents = 0
+    }
+
+    private enum Constants {
+        static let iconProgressSpacing: CGFloat = 12
+        static let progressValueSpacing: CGFloat = 8
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        separatorInset = .init(top: 0, left: UIConstants.horizontalInset, bottom: 0, right: UIConstants.horizontalInset)
+
+        guard progressPercents > 0 else {
+            widthConstraint?.update(offset: 0)
+            progressStackView.setCustomSpacing(0, after: progressView)
+            return
+        }
+        let totalWidth = infoButton.frame.minX
+            - Constants.iconProgressSpacing
+            - iconView.frame.maxX
+            - Constants.iconProgressSpacing
+            - Constants.progressValueSpacing
+            - progressValueLabel.bounds.width
+
+        progressStackView.setCustomSpacing(Constants.progressValueSpacing, after: progressView)
+        let progressViewWidth = totalWidth * CGFloat(progressPercents)
+        widthConstraint?.update(offset: progressViewWidth)
```

Comment:
This might be a problem: `layoutSubviews` is called after auto layout calulations done and here we trigger them again.